### PR TITLE
Updated spec test implementations for v0.8.2 format

### PR DESCRIPTION
**Pull request by shahankhatch**

### CODE REVIEW

File: eth-reference-tests/src/test/java/pegasys/artemis/reference/MapObjectUtil.java [hunk 1]
```diff
@@ -38,14 +38,7 @@
...
 import tech.pegasys.artemis.datastructures.operations.VoluntaryExit;
-import tech.pegasys.artemis.datastructures.state.BeaconState;
-import tech.pegasys.artemis.datastructures.state.Checkpoint;
-import tech.pegasys.artemis.datastructures.state.CompactCommittee;
-import tech.pegasys.artemis.datastructures.state.Crosslink;
-import tech.pegasys.artemis.datastructures.state.Fork;
-import tech.pegasys.artemis.datastructures.state.HistoricalBatch;
-import tech.pegasys.artemis.datastructures.state.PendingAttestation;
-import tech.pegasys.artemis.datastructures.state.Validator;
+import tech.pegasys.artemis.datastructures.state.*;
```
Comment (dangerousfood) [inline eth-reference-tests/src/test/java/pegasys/artemis/reference/MapObjectUtil.java]: errorprone will pick this up. Wildcard imports are not allowed. IntelliJ "helped you" by doing this. Use the following link to disable wildcard imports https://intellij-support.jetbrains.com/hc/en-us/community/posts/206203659-Turn-off-Wildcard-imports-

File: eth-reference-tests/src/test/java/pegasys/artemis/reference/TestSuite.java [hunk 2]
```diff
@@ -47,6 +48,10 @@
           "tests");
   private static final String FILE = "file://";
 
+  //  public static Stream<Arguments> findTests(String glob) throws IOException {
```
Comment (dangerousfood) [inline eth-reference-tests/src/test/java/pegasys/artemis/reference/TestSuite.java]: Are these comments necessary for future updates?

File: eth-reference-tests/src/test/java/pegasys/artemis/reference/TestSuite.java [hunk 3]
```diff
@@ -47,6 +48,10 @@
...
+  //      return Resources.find(glob).map(url -> { return Arguments.of(url); });
+  //  }
+
```
Comment (dangerousfood) [inline eth-reference-tests/src/test/java/pegasys/artemis/reference/TestSuite.java]: Are these comments necessary for future development?

File: eth-reference-tests/src/test/java/pegasys/artemis/reference/TestSuite.java [hunk 4]
```diff
@@ -120,13 +129,19 @@ public static Object pathToObject(Path path) {
...
+  }
+
   @MustBeClosed
   @SuppressWarnings({"unchecked", "rawtypes"})
   public static Stream<Arguments> findTestsByPath(Path path, List<Pair<Class, String>> objectPath)
       throws IOException {
     path = Path.of(pathToTests.toString(), path.toString());
     try (Stream<Path> walk = Files.walk(path)) {
       List<String> result = walk.map(x -> x.toString()).collect(Collectors.toList());
+      //      List<Context> contexts = new ArrayList<>();
```
Comment (dangerousfood) [inline eth-reference-tests/src/test/java/pegasys/artemis/reference/TestSuite.java]: Ditto

File: eth-reference-tests/src/test/java/pegasys/artemis/reference/TestSuite.java [hunk 5]
```diff
@@ -214,4 +247,104 @@ public static Object pathToObject(Path path) {
...
+  @SuppressWarnings({"unchecked", "rawtypes"})
+  @MustBeClosed
+  public static Stream<Arguments> sanitySlotsProcessingSetup(Path path, Path configPath)
+      throws Exception {
+    loadConfigFromPath(configPath);
+
+    List<Pair<Class, String>> arguments = new ArrayList<Pair<Class, String>>();
+    arguments.add(getParams(BeaconStateWithCache.class, "pre.yaml"));
+    arguments.add(getParams(BeaconStateWithCache.class, "post.yaml"));
+    //        arguments.add(getParams(ReadLineType.class, "slots.yaml"));
```
Comment (dangerousfood) [inline eth-reference-tests/src/test/java/pegasys/artemis/reference/TestSuite.java]: Ditto

File: eth-reference-tests/src/test/java/pegasys/artemis/reference/mainnet/phase0/sanity_processing.java [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+      //      tech.pegasys.artemis.datastructures.util.BeaconStateUtil @NotNull
+      //      public static BeaconStateWithCache initialize_beacon_state_from_eth1(Bytes32
+      // eth1_block_hash,
+      //              @NotNull UnsignedLong eth1_timestamp,
+      //              @NotNull List<Deposit> deposits)
+      //      Inferred annotations:
+      //      Method initialize_beacon_state_from_eth1: @org.jetbrains.annotations.NotNull
+      //      Parameter eth1_timestamp: @org.jetbrains.annotations.NotNull
+      //      Parameter deposits: @org.jetbrains.annotations.NotNull
+      //
```
Comment (dangerousfood) [inline eth-reference-tests/src/test/java/pegasys/artemis/reference/mainnet/phase0/sanity_processing.java]: Ditto

File: ethereum/datastructures/src/main/java/tech/pegasys/artemis/datastructures/util/CrosslinkCommitteeUtil.java [hunk 7]
```diff
@@ -138,7 +138,7 @@ public static Integer compute_shuffled_index(int index, int index_count, Bytes32
               get_committee_count(state, epoch).intValue());
 
       // Client specific optimization
-      ((BeaconStateWithCache) state).setCrossLinkCommittee(committee, epoch, shard);
+      //      ((BeaconStateWithCache) state).setCrossLinkCommittee(committee, epoch, shard);
```
Comment (cemozerr) [inline ethereum/datastructures/src/main/java/tech/pegasys/artemis/datastructures/util/CrosslinkCommitteeUtil.java]: Is there any reason we're removing the client-specific optimizations? I haven't looked at state transition in a while, are we just not using them at this version or something?

File: ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/StateTransition.java [hunk 8]
```diff
@@ -170,7 +170,7 @@ private static void process_slot(BeaconState state) {
    * @throws EpochProcessingException
    * @throws SlotProcessingException
    */
-  public static void process_slots(
+  public static BeaconStateWithCache process_slots(
```
Comment (cemozerr) [inline ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/StateTransition.java]: Is there any reason process_slots returns a value here?

File: eth-reference-tests/src/test/java/pegasys/artemis/reference/mainnet/phase0/sanity_processing.java [hunk 9]
```diff
@@ -0,0 +1,167 @@
...
+
+      st.process_slots(spre, blocks.get(0).getSlot(), true);
+
+      st.initiate(spre, blocks.get(0));
+
+      Store genesis_store = ForkChoiceUtil.get_genesis_store(spre);
+      for (int i = 1; i < blocks.size(); i++) {
+        ForkChoiceUtil.on_block(genesis_store, blocks.get(i));
+      }
+      Object a = genesis_store.getFinalized_checkpoint().hash_tree_root();
```
Comment (cemozerr) [inline eth-reference-tests/src/test/java/pegasys/artemis/reference/mainnet/phase0/sanity_processing.java]: More descriptive names for variables would be helpful here

File: ethereum/datastructures/src/main/java/tech/pegasys/artemis/datastructures/util/CrosslinkCommitteeUtil.java (hunk 7 above)
Comment (shahankhatch) [reply ethereum/datastructures/src/main/java/tech/pegasys/artemis/datastructures/util/CrosslinkCommitteeUtil.java]: I thought this was causing me problems, but it's no longer the case (I found the headers were zero hashes and this was not related).

Comment (shahankhatch) [reply ethereum/datastructures/src/main/java/tech/pegasys/artemis/datastructures/util/CrosslinkCommitteeUtil.java]: I'm adding it back in.

File: ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/StateTransition.java (hunk 8 above)
Comment (shahankhatch) [reply ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/StateTransition.java]: I was just copying the return from the other methods, which allows for 'fluent' design by returning the input parameter. I can revert to void if that's preferred.

File: eth-reference-tests/src/test/java/pegasys/artemis/reference/mainnet/phase0/sanity_processing.java (hunk 9 above)
Comment (shahankhatch) [reply eth-reference-tests/src/test/java/pegasys/artemis/reference/mainnet/phase0/sanity_processing.java]: done

File: ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/StateTransition.java (hunk 8 above)
Comment (shahankhatch) [reply ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/StateTransition.java]: done

File: ethereum/datastructures/src/main/java/tech/pegasys/artemis/datastructures/util/CrosslinkCommitteeUtil.java (hunk 7 above)
Comment (shahankhatch) [reply ethereum/datastructures/src/main/java/tech/pegasys/artemis/datastructures/util/CrosslinkCommitteeUtil.java]: done

File: eth-reference-tests/src/test/java/pegasys/artemis/reference/mainnet/phase0/sanity_processing.java (hunk 6 above)
Comment (shahankhatch) [reply eth-reference-tests/src/test/java/pegasys/artemis/reference/mainnet/phase0/sanity_processing.java]: done

File: eth-reference-tests/src/test/java/pegasys/artemis/reference/TestSuite.java (hunk 5 above)
Comment (shahankhatch) [reply eth-reference-tests/src/test/java/pegasys/artemis/reference/TestSuite.java]: done

File: eth-reference-tests/src/test/java/pegasys/artemis/reference/TestSuite.java (hunk 4 above)
Comment (shahankhatch) [reply eth-reference-tests/src/test/java/pegasys/artemis/reference/TestSuite.java]: done

File: eth-reference-tests/src/test/java/pegasys/artemis/reference/TestSuite.java (hunk 3 above)
Comment (shahankhatch) [reply eth-reference-tests/src/test/java/pegasys/artemis/reference/TestSuite.java]: remove. done

File: eth-reference-tests/src/test/java/pegasys/artemis/reference/MapObjectUtil.java (hunk 1 above)
Comment (shahankhatch) [reply eth-reference-tests/src/test/java/pegasys/artemis/reference/MapObjectUtil.java]: thanks, updated my settings, fixed the import.

File: eth-reference-tests/src/test/java/pegasys/artemis/reference/TestSuite.java [hunk 10]
```diff
@@ -16,9 +16,7 @@
 import com.fasterxml.jackson.databind.ObjectMapper;
 import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
 import com.google.errorprone.annotations.MustBeClosed;
-import java.io.IOException;
-import java.io.InputStream;
-import java.io.UncheckedIOException;
+import java.io.*;
```
Comment (dangerousfood) [inline eth-reference-tests/src/test/java/pegasys/artemis/reference/TestSuite.java]: Wildcard import

File: eth-reference-tests/src/test/java/pegasys/artemis/reference/TestSuite.java [hunk 11]
```diff
@@ -34,7 +32,10 @@
 import org.apache.tuweni.io.Resources;
 import org.junit.jupiter.params.provider.Arguments;
 import tech.pegasys.artemis.datastructures.Constants;
+import tech.pegasys.artemis.datastructures.blocks.BeaconBlock;
+import tech.pegasys.artemis.datastructures.operations.*;
```
Comment (dangerousfood) [inline eth-reference-tests/src/test/java/pegasys/artemis/reference/TestSuite.java]: Wildcard import

File: eth-reference-tests/src/test/java/pegasys/artemis/reference/mainnet/phase0/operations_processing.java [hunk 12]
```diff
@@ -0,0 +1,134 @@
...
+import java.util.List;
+import java.util.stream.Stream;
+import org.apache.tuweni.junit.BouncyCastleExtension;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import pegasys.artemis.reference.TestSuite;
+import tech.pegasys.artemis.datastructures.blocks.BeaconBlock;
+import tech.pegasys.artemis.datastructures.operations.*;
```
Comment (dangerousfood) [inline eth-reference-tests/src/test/java/pegasys/artemis/reference/mainnet/phase0/operations_processing.java]: Wildcard import

Comment (shahankhatch) [reply eth-reference-tests/src/test/java/pegasys/artemis/reference/mainnet/phase0/operations_processing.java]: done

File: eth-reference-tests/src/test/java/pegasys/artemis/reference/TestSuite.java (hunk 11 above)
Comment (shahankhatch) [reply eth-reference-tests/src/test/java/pegasys/artemis/reference/TestSuite.java]: done

File: eth-reference-tests/src/test/java/pegasys/artemis/reference/TestSuite.java (hunk 10 above)
Comment (shahankhatch) [reply eth-reference-tests/src/test/java/pegasys/artemis/reference/TestSuite.java]: done

**Review by dangerousfood — APPROVED** [review]
LGTM

**Pull request merged**